content/posts/unpublished/backbase-tech-stack/post.md

```diff
@@ -0,0 +1,198 @@
+# Backbase tech stack
+
+What exactly is being used to build banking applications in Backbase?
+
+![](assets/tech_stack_img_2.jpeg)
+
+Authors: Paweł Garbacik
+Date: unpublished
+Category: tech-life
+
+tags: architecture,technology,patterns,stack
+
+---
+
+## Introduction
+
+Backbase Engagement Banking Services is a comprehensive solution with a lot of capabilities.
+It is thought out and designed to provide modern architecture with the latest market standards,
+the most advanced security mechanisms, and great extensibility and flexibility to make it easy
+to integrate with different core banking systems.
+
+To achieve this, different technologies, frameworks, standards, and architectural patterns need to be combined.
+
+
+This article aims to provide an insight into how the Backbase software is built and integrated on a high level,
+what kind of technologies are used on a different levels and layers.
+
+## High level architecture
+
+<img align="right" src="assets/HL_Architecture.drawio.png">
+
+Engagement Banking Services are divided according to the capabilities they offer, the role they play, and on which level they operate.
+
+Each service is designed to perform completely different tasks and offers specific ways to interact with it.
+
+On the top there are client-facing apps. These applications are available for 2 different platforms:
+
+* Web applications - available on modern web-browsers, dedicated for different group of users, segregated by the roles and responsibilities
+* Mobile applications - available on popular mobile devices powered by iOS and Android Mobile OSs
+
+These applications allow to interact with backend services for these users who have proper permissions.
+Client APIs are a dedicated and well secured APIs designed for interaction between client-facing applications and backend services.
+
+On the next layer, there are services which take care of the security of the client-facing applications.
+These layer of services is able to recognize the user and introduce the same user to backend services in a more understandable way.
+This layer also offers the set of generic services which delivers generic capabilities like document storage or auditing for all other specific services.
+
+Next layer offers specific banking services like Arrangements, Batches, Cards, Contacts, Payments etc.
```

Comment:
this entire section is a bit verbose, I'm not sure how valuable it is when it comes to elucidate of the tech stack itself

Reply:
Changed and simplified this text